pkgs/applications/video/screenkey/default.nix

```diff
@@ -1,53 +1,65 @@
 { lib
-, substituteAll
-, buildPythonApplication
-, fetchFromGitHub
-, distutils_extra
-, setuptools-git
+, fetchFromGitLab
+# native
 , intltool
-, pygtk
-, libX11
-, libXtst
 , wrapGAppsHook
-, gnome3
+, file
+# not native
+, xorg
+, gobject-introspection
+, gtk3
+, python3
 }:
-buildPythonApplication rec {
+
+python3.pkgs.buildPythonApplication rec {
   pname = "screenkey";
-  version = "0.9";
+  version = "1.2";
 
-  src = fetchFromGitHub {
-    owner = "wavexx";
+  src = fetchFromGitLab {
+    owner = "screenkey";
     repo = "screenkey";
-    rev = "screenkey-${version}";
-    sha256 = "14g7fiv9n7m03djwz1pp5034pffi87ssvss9bc1q8vq0ksn23vrw";
+    rev = "v${version}";
+    sha256 = "1x13n57iy2pg3h3r994q3g5nbmh2gwk3qidmmcv0g7qa89n2gwbj";
   };
 
-  patches = [
-    (substituteAll {
-      src = ./paths.patch;
-      inherit libX11 libXtst;
-    })
-  ];
-
   nativeBuildInputs = [
-    distutils_extra
-    setuptools-git
+    python3.pkgs.distutils_extra
+    # Shouldn't be needed once https://gitlab.com/screenkey/screenkey/-/issues/122 is fixed.
     intltool
-
+    # We are not sure why is this needed, but without it we get "file: command
+    # not found" errors during build.
+    file
+    # This is here and not in buildInputs to avoid scrictDeps issue:
+    # https://github.com/NixOS/nixpkgs/issues/56943
+    gobject-introspection
```

Comment:
It is actually the other way around. It belongs here but causes `strictDeps` issues. One way to “fix” it is to move it to `buildInputs` as you did before but that is a hack that has its own issues. I prefer disabling `strictDeps` instead, which is slightly less hacky.

Reply:
I'm not sure I got to the bottom of this issue, but I moved `gobject-introspection` to `buildInputs` and disabled `strictDeps`.

Reply:
I meant that it should remain in `nativeBuildInputs` (since that is the correct place) and `strictDeps` should be disabled (to bypass the bug).

Reply:
> I meant that it should remain in nativeBuildInputs (since that is the correct place)

Really? Why?

Also, I didn't experience any issue running `screenkey` with either of the options I tested, with and without `scrictDeps` set. Still, now everything should be in place as you wish.

Reply:
We add `gobject-introspection` for a setup hook so it should go to `nativeBuildInputs`. But that is currently broken with `strictDeps = true` (default in Python packages).